tests/ops/integration_tests/saas/test_shopify_task.py

```diff
@@ -260,3 +262,247 @@ async def test_shopify_access_request_task(
+            "total_price",
+            "total_price_usd",
+            "total_tax",
+            "total_tip_received",
+            "total_weight",
+            "updated_at",
+            "user_id",
+            "customer",
+            "discount_applications",
+            "payment_terms",
+        ],
+    )
+
+    assert_rows_match(
+        v[f"{dataset_name}:customer_addresses"],
+        min_size=1,
+        keys=[
+            "id",
+            "customer_id",
+            "first_name",
+            "last_name",
+            "company",
+            "address1",
+            "address2",
+            "city",
+            "province",
+            "country",
+            "zip",
+            "phone",
+            "name",
+            "province_code",
+            "country_code",
+            "country_name",
+            "default",
+        ],
+    )
+
+    assert_rows_match(
+        v[f"{dataset_name}:blogs"],
+        min_size=1,
+        keys=[
+            "id",
+            "handle",
+            "title",
+            "updated_at",
+            "commentable",
+            "feedburner",
+            "feedburner_location",
+            "created_at",
+            "template_suffix",
+            "tags",
+            "admin_graphql_api_id",
+        ],
+    )
+
+    assert_rows_match(
+        v[f"{dataset_name}:blog_articles"],
+        min_size=1,
+        keys=[
+            "id",
+            "title",
+            "created_at",
+            "body_html",
+            "blog_id",
+            "author",
+            "user_id",
+            "published_at",
+            "updated_at",
+            "summary_html",
+            "template_suffix",
+            "handle",
+            "tags",
+            "admin_graphql_api_id",
+        ],
+    )
+
+    assert_rows_match(
+        v[f"{dataset_name}:blog_article_comments"],
+        min_size=1,
+        keys=[
+            "id",
+            "body",
+            "body_html",
+            "author",
+            "email",
+            "status",
+            "article_id",
+            "blog_id",
+            "created_at",
+            "updated_at",
+            "ip",
+            "user_agent",
+            "published_at",
+        ],
+    )
+
+    temp_masking = config.execution.masking_strict
+    config.execution.masking_strict = True
+
+    x = await graph_task.run_erasure(
+        privacy_request,
+        erasure_policy_string_rewrite,
+        graph,
+        [shopify_connection_config],
+        {"email": shopify_erasure_identity_email},
+        get_cached_data_for_erasures(privacy_request.id),
+        db,
+    )
+
+    assert x == {
+        f"{dataset_name}:customers": 1,
+        f"{dataset_name}:blogs": 0,
+        f"{dataset_name}:customer_orders": 1,
+        f"{dataset_name}:customer_addresses": 1,
+        f"{dataset_name}:blog_articles": 0,
+        f"{dataset_name}:blog_article_comments": 1,
+        f"{dataset_name}:customer_order_transactions": 0,
+    }
```

Comment:
i assume that the updates take too long on the source system (i.e. on the shopify platform) to actually query to make sure our data gets updated - but just want to check.

Reply:
So here we were first going for only getting existing data in erasure fixture instead of creating new(like we are doing for blogs endpoint). Since after first test data was masked already, quering it again to check if it is masked didn't make sense at that time that is why we didn't add actual query check. Since now as we have changed our approach after discussion and we are actually creating some new data(and deleting it after fixture) it'll be good to add actual query checks, I'll add it now.